python/ray/serve/backend_state.py

```diff
@@ -441,19 +444,34 @@ def scale_backend_replicas(
                     or replica_state_dict[ReplicaState.STARTING] \
                     or replica_state_dict[ReplicaState.RUNNING]
 
-                if not len(list_to_use):
-                    assert False, replica_state_dict
+                assert len(list_to_use), replica_state_dict
                 replica_to_stop = list_to_use.pop()
 
                 graceful_timeout_s = (backend_info.backend_config.
                                       experimental_graceful_shutdown_timeout_s)
-                if force_kill:
-                    graceful_timeout_s = 0
 
                 replica_to_stop.set_should_stop(graceful_timeout_s)
                 self._replicas[backend_tag][ReplicaState.SHOULD_STOP].append(
                     replica_to_stop)
 
+        return True
+
+    def scale_all_backends(self):
+        checkpoint_needed = False
+        for backend_tag, num_replicas in list(self._target_replicas.items()):
+            try:
+                checkpoint_needed = (checkpoint_needed or self.scale_backend_replicas(backend_tag))
+            except RayServeException as e:
+                del self._backend_metadata[backend_tag]
+                del self._target_replicas[backend_tag]
+                raise e
```

Comment:
what is this for? it doesn't seem like we want to do this in every case that scaling fails

Reply:
I don't fully understand what the initial intent of the codepath was. It was checking for 'infeasible placements', but it was only being checked on create a backend, not scale a backend.

Reply:
Oh I see, this shouldn't be relevant anymore then. I'd remove it.

Reply:
sgtm